src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -1000,12 +1025,13 @@ muse::Ret Au3Interaction::pasteFromClipboard(secs_t begin, bool moveClips, bool
     }
 
     muse::Ret ok { muse::make_ok() };
+    bool pasteIntoExistingClip = !configuration()->pasteAsNewClip() && !moveAllTracks;
```

Comment:
I would prefer `Au3Interaction` not to depend on the configuration. It should only rely on arguments passed to the function. That way it would be easier to test and use in scripting.

Reply:
as a quick and dirty fix we could add a simple struct with a few bools, or refactor the function (not sure if this is a good idea though)

Reply:
Au3Interaction is a simple service, it is no different from our other services, so I see no limitations in using the configurator.

As for scripts, we can set up the configurator before using this service. For example, we already do this - parsing command line arguments (applyCommandLineOptions). When a user calls a program with a flag, we simply set this value in the configurator and then the service works with this data.

On the other hand, we stick to simple interfaces for this kind of services, with a minimum number of parameters, so it is easier to understand what the method does without delving into the code of this method.

Reply:
>so it is easier to understand what the method does without delving into the code of this method.

I mean it is harder to understand what and why the method does if its behavior can change depending on the arguments and the configuration simultaneously.

Reply:
A classic example of a debate about simplicity of interfaces and predictability of code behavior.

Your approach is more related to functional programming, my approach is more related to object-oriented programming.

My opinion is that the second approach is more suitable for such large products as Audacity. The less code, the better. The case with scripts is a special case, we should not change all our approaches because of this limitation. Just because it can be called from scripts, we have to add a bunch of parameters, structures, check their validity, think about their default values? I think this is a complication.
I think it is easier to hide such settings inside the service and only provide externally those parameters that are really needed.